source/Mlos.Python/mlos/Spaces/HypergridAdapters/HypergridAdapter.py

```diff
@@ -113,7 +113,7 @@ def _project_point(self, point: Point) -> Point:
         :return:
         """
         original_dataframe = point.to_dataframe()
-        projected_dataframe = self.project_dataframe(original_dataframe)
+        projected_dataframe = self._project_dataframe(original_dataframe)

```

Comment:
This is another bug-fix. Thank you for including it in this PR. A few words of explanation for the benefit of other reviewers:
* project_dataframe (public, without leading underscore) recursively invokes all adapters in the stack. For example: it might first invoke HierarchicalToFlatHypergridAdapter, then the CategoricalToOneHotEncodedHypergridAdapter. 
* _project_dataframe (protected, with leading underscore) executes only the current adapter's projection. 

Similar logic applies to: project_point and _project_point.

Thus, inside of the `_project_point()` function, we wish to only perform the projection defined by `self`, which is accomplished by calling transforming a point to a dataframe, performing self._project_dataframe(), and then recovering the point from the dataframe.